ui/src/components/PlaylistPage.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router";
+import NavBar from "./NavBar";
+import { PlaylistTracks } from "../types";
+import PlayCircleRoundedIcon from "@mui/icons-material/PlayCircleRounded";
+import FavoriteBorderOutlinedIcon from "@mui/icons-material/FavoriteBorderOutlined";
+import { green, red } from "@mui/material/colors";
+import Auth from "./Auth";
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export default function PlaylistPage() {
+  const { id } = useParams();
+  const [playlistName, setPlaylistName] = useState("");
+  const [playlistTracks, setPlaylistTracks] = useState<PlaylistTracks[]>([]);
+  const [playlistImage, setPlaylistImage] = useState("");
+  const [playlistInfo, setPlaylistInfo] = useState("");
+  const [playlistFollowers, setPlaylistFollowers] = useState("");
+  const [owner, setOwner] = useState("");
```

Comment:
This could be improved by replacing all the states with a single state which stores an object